include/conf/defaults.hpp

```diff
@@ -70,24 +72,34 @@ namespace color_coded
         "-isystem/opt/local/include",
         environment<env::tag>::clang_resource_dir, // internal libraries and intrinsics
         "-isystem/usr/include",
+#ifdef __APPLE__
         "-isystem/System/Library/Frameworks",
         "-isystem/Library/Frameworks",
+#endif
         "-w",
         "-fcolor-diagnostics" // See https://github.com/jeaye/color_coded/issues/104
       };
     }
 
+    /* Add system defaults to user specified arguments. Needed because libclang
+     * often fails to find system search paths. */
+    args_t add_defaults_to_args(std::string const &filetype, args_t &args)
```

Comment:
Indeed it's a confusing interface.

I actually started out with `void add_defaults_to_args(std::string const &filetype, args_t &args)` and modified `args` directly. Yet I thought `insert` a range of items into `vector` would have quadratic complexity so I tried to avoid it. Hence the return value.

Now that I realized that `insert` would only move items once if `InsertEndIter - InsertStartIter` is known, perhaps it's better to switch back to void instead of using move-in-move-out. What do you think?